runtime/include/chpl-comm-diags.h

```diff
@@ -29,32 +32,68 @@ typedef struct _chpl_atomic_commDiagnostics {
 #undef _COMM_DIAGS_DECL_ATOMIC
 } chpl_atomic_commDiagnostics;
 
+chpl_atomic_commDiagnostics chpl_comm_diags;
```

Comment:
By moving this into a header, won't we have different instance for each .c file that includes it? That's not a problem for ugni/gasnet right now since they're monolithic files, but if we ever split ugni or use this in multiple files in ofi, is this what we want?

Reply:
That would be true if it there were a prior declaration with the `static` storage class specifier; it would have so-called internal linkage and each declaration would create a unique object.  But since there is no `static` declaration of it anywhere, declaring it without a storage class specifier is the same as declaring it with an explicit `extern`, and it has external linkage: all the declarations name the same object.  See the C standard, section 6.2.2.  But I'll change it to an explicit `extern` declaration to make this more clear.  (The various variable decls in `chpl-comm.h` are this way.)

Reply:
Hmm, adding the `extern` storage class to the declaration in the .h file actually means that we have to add another declaration without any storage class in one single place somewhere (maybe `chpl-comm.c`, for example).  Without that we get undefined externs at link time, which I hadn't expected from reading the standard.  I'm good with either doing that or leaving off the `extern` in the `.h` file which, will have the same effect.  Do you have a preference?

In the meantime I'm also going to rename `chpl_comm_diags`, perhaps to `chpl_comm_diags_counters`, for clarity because it's too similar to the existing `chpl_comm_diagnostics` flag for my taste.

Reply:
My preference would be to leave the `extern` storage class specifier off entirely, since then the variables only need to be mentioned in `chpl-comm-diags.h`, nowhere else.  (Switched to plural because whatever we do for the existing `chpl_comm_diags` also has to be done for `chpl_comm_diags_disable_flag`.)

Reply:
Works for me